### PR TITLE
I added the new tool query_IMPC

### PR DESCRIPTION
**Pull request by AndreaFurlani**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/query_IMPC/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: query_IMPC
```
Comment (bgruening) [inline tools/query_IMPC/.shed.yml:1]: lower case names please

File: tools/query_IMPC/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+owner: infrafrontier
```
Comment (bgruening) [inline tools/query_IMPC/.shed.yml:2]:
```suggestion
owner: iuc
```

File: tools/query_IMPC/.shed.yml (hunk 1 above)
Comment (bgruening) [reply tools/query_IMPC/.shed.yml:1]:
```suggestion
name: query_impc
```

File: tools/query_IMPC/impc_tool.xml [hunk 3]
```diff
@@ -0,0 +1,365 @@
+<tool id="IMPC_query_tool" name="IMPC" version="0.9.0">
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:1]:
```suggestion
<tool id="query_impc" name="IMPC" version="0.9.0" profile="22.05">
```

File: tools/query_IMPC/impc_tool.xml [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+  <description> query tool.</description>
+  <requirements>
+    <requirement type="package" version="2.25.1">requests</requirement>
+    <requirement type="package" version="1.3.5">pandas</requirement>
+	<requirement type="package" version="4.9.2">lxml</requirement>
+	<requirement type="package" version="3.2.2">mygene</requirement>
+  </requirements>
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:8]:
```suggestion
  <requirements>
        <requirement type="package" version="2.25.1">requests</requirement>
        <requirement type="package" version="1.3.5">pandas</requirement>
	<requirement type="package" version="4.9.2">lxml</requirement>
	<requirement type="package" version="3.2.2">mygene</requirement>
  </requirements>
```

File: tools/query_IMPC/impc_tool.xml [hunk 5]
```diff
@@ -0,0 +1,365 @@
+<tool id="IMPC_query_tool" name="IMPC" version="0.9.0">
+  <description> query tool.</description>
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:2]:
```suggestion
  <description>query tool</description>
```

File: tools/query_IMPC/impc_tool.xml [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+  <requirements>
+    <requirement type="package" version="2.25.1">requests</requirement>
+    <requirement type="package" version="1.3.5">pandas</requirement>
+	<requirement type="package" version="4.9.2">lxml</requirement>
+	<requirement type="package" version="3.2.2">mygene</requirement>
+  </requirements>
+  <stdio>
+  </stdio>
+  <command>
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:11]:
```suggestion
    <command detect_errors="exit_code">
```

File: tools/query_IMPC/impc_tool.xml [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+	  #if $query_type.selector == '7'
+	  	'$query_type.input' '$output' '$query_type.selector' '$query_type.head' '$query_type.g_out'
+	  #else
+	  	#if $query_type.inp_q.inp_sel == 'str'
+	    	'$query_type.inp_q.input' '$output' '$query_type.selector' '$query_type.head' '$query_type.inp_q.inp_sel'
+		#else
+			'$query_type.inp_q.input' '$output' '$query_type.selector' '$query_type.head' '$query_type.inp_q.inp_sel' '$query_type.inp_q.sep'
+		#end if
+	  #end if
+	  #if $query_type.selector == '2' or $query_type.selector == '8' or $query_type.selector == '9'
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:23]:
```suggestion
	  #if $query_type.selector in ['2', '8', '9']:
```

File: tools/query_IMPC/impc_tool.xml [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+	  	'$query_type.g_out'
+	  #end if]]>
+  </command>
+  <inputs>
+			<conditional name="query_type">
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:28]: please fix the indentation here.
I recommend you to use https://github.com/galaxyproject/galaxy-language-server

File: tools/query_IMPC/impc_tool.xml [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+					  <conditional name="inp_q">
+						  <param name="inp_sel" type="select" label="Select the type of input">
+							  <option value="str">1 - Direct input</option>
+							  <option value="txt">2 - Txt file</option>
+						  </param>
+						  <when value="str">
+							  <param name="input" type="text" label="Input gene:" help="Enter a single MGI gene id or gene symbol"/>
+						  </when>
+						  <when value="txt">
+							  <param name="input" type="data" format="data,tabular,txt" label="Input file:" help="Enter a txt file with the Gene MGI id or gene symbol"/>
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:50]:
```suggestion
							  <param name="input" type="data" format="tabular,txt" label="Input file:" help="Enter a txt file with the Gene MGI id or gene symbol"/>
```

File: tools/query_IMPC/impc_tool.xml [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+							  <param name="sep" type="select" label="Select the separator used in the file:">
+								  <option value="t">tab</option>
+								  <option value="s">single space</option>
+								  <option value=",">,</option>
+								  <option value=";">;</option>
+							  </param>
+						  </when>
+					  </conditional>
+						  <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:59]: Do you know the https://docs.galaxyproject.org/en/latest/dev/schema.html#data-column parameter? Maybe that helps you in this tool.

File: tools/query_IMPC/impc_tool.xml [hunk 11]
```diff
@@ -0,0 +1,365 @@
...
+						  <param name="inp_sel" type="select" label="Select the type of input">
+							  <option value="str">1 - Direct input</option>
+							  <option value="txt">2 - Txt file</option>
+						  </param>
+						  <when value="str">
+                          		<param name="input" type="text" label="Input id:" help="Enter a top level phenotype category id"/>
+						  </when>
+						  <when value="txt">
+							  <param name="input" type="data" format="data,tabular,txt" label="Input file:" help="Enter a txt file with a top level phenotype category id"/>
+							  <param name="sep" type="select" label="Select the separator used in the file:">
```
Comment (bgruening) [inline tools/query_IMPC/impc_tool.xml:214]: You are repeating a lot of code here. Please consider to use macros.

**Review by bgruening — COMMENTED** [review]
Thanks @AndreaFurlani!

A few initial comments are inline.

File: tools/query_IMPC/impc_tool.xml (hunk 10 above)
Comment (AndreaFurlani) [reply tools/query_IMPC/impc_tool.xml:59]: Thank you for the suggestion, we will try to use it for the second iteration

File: tools/query_IMPC/impc_tool.xml (hunk 11 above)
Comment (AndreaFurlani) [reply tools/query_IMPC/impc_tool.xml:214]: Thank you, also for this we will use it in the second iteration

**Pull request closed without merging**